### PR TITLE
Blank lookup in a search should produce error

### PR DESCRIPTION
**Pull request by brainopia**
Example
```
search
  lookup[]
```

Will produce error

```
Searching a lookup requires at least one of record, attribute, value, or node to be present
```

_(text may need a proofreading, since I'm not a native English speaker)_

I'll probably send another PR tomorrow to fix `lookup` to work with partially missing variables (related  https://github.com/witheve/Eve/pull/611#issuecomment-263498342).

---
This change is [Reviewable](https://reviewable.io/reviews/witheve/eve/619)

### CODE REVIEW

File: src/runtime/join.ts [hunk 1]
```diff
@@ -218,6 +218,109 @@ export class Scan {
     return foo.join("");
   }
 
+  
+  // // I can write macros with eval, hmm
```
Comment (cmontella) [inline src/runtime/join.ts]: I think that if this commented code is part of other work, it should be included in a separate PR. For now just take the commented code out of this PR.

File: test/join.ts [hunk 2]
```diff
@@ -1907,6 +1907,22 @@ test("pipe allows you to select ", (assert) => {
...
 })
 
+test("blank lookup errors", (assert) => {
+  let expected = {
+    insert: [],
+    remove: [],
+    errors: true
+  };
+  evaluate(assert, expected, `
+    foo
```
Comment (cmontella) [inline test/join.ts]: No need for this foo

File: src/runtime/errors.ts [hunk 3]
```diff
@@ -257,6 +263,7 @@ export var messages = {
 
   unimplementedExpression: (op) => `There's no definition for the function ${op}`,
 
+  blankScan: () => 'Searching a lookup requires at least one of record, attribute, value, or node to be present',
```
Comment (cmontella) [inline src/runtime/errors.ts]: I would change the error to:

"lookup requires at least one attribute: record, attribute, value, or node"

File: test/shared_functions.ts [hunk 4]
```diff
@@ -151,6 +151,8 @@ export function evaluate(assert, expected, code, session = new Database()) {
   let {blocks, errors} = builder.buildDoc(parsed.results);
   if(expected.errors) {
     assert.true(parsed.errors.length > 0 || errors.length > 0, "This test is supposed to produce errors");
+  } else {
+    assert.true(parsed.errors.length == 0 && errors.length == 0, "This test is not supposed to produce errors");
```
Comment (cmontella) [inline test/shared_functions.ts]: We just merged in changes to shared_functions.ts... this change should merge in fine, but it could cause a conflict. We'll see.

**Comment by brainopia** [comment]
Oops, changes in `join.ts` and `shared_functions.ts` were incidentally in this commit. Seems like when I've decided to re-push commit with amended title, I've accidentally added a few extra things.

All fixed now, thank you!

**Comment by ibdknox** [comment]
Thanks!
